File: majorProject/src/main/java/com/aocc/majorproject/ui/UiLayout.java
package com.aocc.majorproject.ui;

import com.aocc.framework.GameConstants;

/** Helpers for placing UI within the 1280×720 world rectangle. */
public final class UiLayout {

    private UiLayout() {
    }

    public static int centerX(int width) {
        return (GameConstants.WORLD_WIDTH - width) / 2;
    }

    public static int centerY(int height) {
        return (GameConstants.WORLD_HEIGHT - height) / 2;
    }

    public static int alignRight(int width, int margin) {
        return GameConstants.WORLD_WIDTH - width - margin;
    }
}

File: majorProject/src/main/kotlin/com/aocc/majorproject/ui/UiLayout.kt
package com.aocc.majorproject.ui

import com.aocc.framework.GameConstants

/** Helpers for placing UI within the 1280×720 world rectangle. */
object UiLayout {

    @JvmStatic
    fun centerX(width: Int): Int {
        return (GameConstants.WORLD_WIDTH - width) / 2
    }

    @JvmStatic
    fun centerY(height: Int): Int {
        return (GameConstants.WORLD_HEIGHT - height) / 2
    }

    @JvmStatic
    fun alignRight(width: Int, margin: Int): Int {
        return GameConstants.WORLD_WIDTH - width - margin
    }
}
